lib/ansible/modules/cloud/ovirt/ovirt_vms.py

```diff
@@ -1317,6 +1337,39 @@ def __get_vnic_profile_id(self, nic):
                 )
             )
 
+    def __attach_numa_node(self, entity):
+        numa_nodes_service = self._service.service(entity.id).numaNodesService()
+        numa_node = self.param('numa_node')
+
+        if numa_node is None or numa_node.get('index') is None:
+            return False
+
+        for current_numa_node in numa_nodes_service.list():
+            if current_numa_node.get('index') is not None and current_numa_node.get('index') == numa_node.get('index'):
+                return False
+
+        numa_nodes_service.add(
+            otypes.VirtualNumaNode(
+                index=otypes.Index(numa_node.get('index')),
+                memory=otypes.Memory(numa_node.get('memory')),
+                numa_node_pins=[
+                    otypes.numa_node_pins(
+                        otypes.NumaNodePin(
+                            index=otypes.index(numa_node.get('numa_node_pin'))
+                        )
+                    ),
+                ] if numa_node.get('numa_node_pin') is not None else None,
+                cpu=otypes.Cpu(
+                    cores=[
+                        otypes.Cores(
+                            index=otypes.index(core)
```

Comment:
otypes.Core(index=core) for core in numa_node.get('cores')

Reply:
Done